.github/workflows/publish_maven_central.yml

```
@@ -0,0 +1,39 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Publish package to Maven central
+
+on:
+  push:
+    branches:
+      - mpolak/gh-action-publish-maven-central
+    tags: v*
+
+jobs:
+  publish:
+    runs-on: [self-hosted, zendesk-stable]
+
+    steps:
+    - uses: zendesk/checkout@v2
+    - name: Set up JDK 1.8
+      uses: zendesk/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Prepare environment
+      env:
+        GPG_KEY_CONTENTS: ${{ secrets.GPG_KEY_CONTENTS }}
+        SIGNING_SECRET_KEY_RING_FILE: ~/.keyring.gpg
+      run: |
+        git fetch --unshallow
+        echo '$GPG_KEY_CONTENTS' | base64 -d > '$SIGNING_SECRET_KEY_RING_FILE'
```

Single quotes won't substitute the ENV vars :D 

```suggestion
        git fetch --unshallow
        echo "$GPG_KEY_CONTENTS" | base64 -d > "$SIGNING_SECRET_KEY_RING_FILE"
```